GetStream/stream-chat-android
# [3754] fix issue with a socket being stuck

Closes #3754

### 🎯 Goal

The socket should reconnect when the network becomes available.

### 🧪 Testing

Steps:
- enable airplane mode
- wait for a Disconnected state
- background app
- foreground app
- disable airplane mode

Make sure socket gets connected

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_
